frida-gum/src/debug_symbol.rs

```diff
@@ -0,0 +1,75 @@
+use std::borrow::Borrow;
+use std::convert::TryInto;
+use std::fmt;
+use std::mem::MaybeUninit;
+use std::{ffi::CStr, str::Utf8Error};
+
+use frida_gum_sys as gum_sys;
+use gum_sys::{gum_symbol_details_from_address, GumDebugSymbolDetails};
+
+use crate::NativePointer;
+
+pub struct Symbol {
+    gum_debug_symbol_details: GumDebugSymbolDetails,
+}
+
+impl fmt::Debug for Symbol {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("Symbol")
+            .field("address", &self.address())
+            .field("module_name", &self.module_name())
+            .field("symbol_name", &self.symbol_name())
+            .field("file_name", &self.file_name())
+            .field("line_number", &self.line_number())
+            .finish()
+    }
+}
+
+impl Symbol {
+    /// Address that this symbol is for.
+    pub fn address(&self) -> usize {
+        self.gum_debug_symbol_details.address.try_into().unwrap()
+    }
+
+    /// Name of the symbol, as a string.
+    pub fn module_name(&self) -> Result<&str, Utf8Error> {
+        unsafe { CStr::from_ptr(self.gum_debug_symbol_details.module_name.as_ptr()) }.to_str()
+    }
+
+    /// Module name owning this symbol, as a string.
+    pub fn symbol_name(&self) -> Result<&str, Utf8Error> {
+        unsafe { CStr::from_ptr(self.gum_debug_symbol_details.symbol_name.as_ptr()) }.to_str()
+    }
+
+    /// File name owning this symbol, as a string.
+    pub fn file_name(&self) -> Result<&str, Utf8Error> {
+        unsafe { CStr::from_ptr(self.gum_debug_symbol_details.file_name.as_ptr()) }.to_str()
+    }
+
+    /// Line number in file_name, as a number.
+    pub fn line_number(&self) -> u32 {
+        self.gum_debug_symbol_details.line_number
+    }
```

Comment:
It's not necessary to include `as a number` or `as a string` because the type annotations are available in the documentation.

Reply:
Fixed in fd9de4ba5fe95500a20f8fdd70da88e9748f5bb1